src/backend/cpu/diagonal.cpp

```diff
@@ -36,9 +38,9 @@ Array<T> diagCreate(const Array<T> &in, const int num) {
 
 template<typename T>
 Array<T> diagExtract(const Array<T> &in, const int num) {
-    const dim4 idims = in.dims();
-    dim_t size       = std::min(idims[0], idims[1]) - std::abs(num);
-    Array<T> out     = createEmptyArray<T>(dim4(size, 1, idims[2], idims[3]));
+    const dim4 &idims = in.dims();
+    dim_t size        = min(idims[0], idims[1]) - abs(num);
```

Comment:
std::min and std::max will have the same problems on windows since those are defined as a macro. We guard against those using the LEAN_AND_MEAN defintion we pass to the compiler. 

namespace should only be used in case it is ambiguous to compile. Here its not.